libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -92,72 +92,94 @@ void AC_PrecLand::update(float alt_above_terrain_cm)
     }
 }
 
-// get_target_shift - returns 3D vector of earth-frame position adjustments to target
-Vector3f AC_PrecLand::get_target_shift(const Vector3f &orig_target)
+bool AC_PrecLand::target_acquired()
 {
-    Vector3f shift; // default shift initialised to zero
+    return (AP_HAL::millis()-_last_update_ms) < 1000;
+}
 
-    // do not shift target if not enabled or no position estimate
-    if (_backend == NULL || !_have_estimate) {
-        return shift;
+bool AC_PrecLand::get_target_position(Vector3f& ret)
+{
+    if (!target_acquired()) {
+        return false;
     }
 
-    // shift is target_offset - (original target - current position)
-    Vector3f curr_offset_from_target = orig_target - _inav.get_position();
-    shift = _target_pos_offset - curr_offset_from_target;
-    shift.z = 0.0f;
+    ret = _target_pos;
+    return true;
+}
+
+bool AC_PrecLand::get_target_position_relative(Vector3f& ret)
+{
+    if (!target_acquired()) {
+        return false;
+    }
 
-    // record we have consumed this reading (perhaps there is a cleaner way to do this using timestamps)
-    _have_estimate = false;
+    ret = _target_pos_rel;
+    return true;
+}
 
-    // return adjusted target
-    return shift;
+bool AC_PrecLand::get_target_velocity_relative(Vector3f& ret)
+{
+    return false;
 }
 
-// calc_angles_and_pos - converts sensor's body-frame angles to earth-frame angles and position estimate
+// converts sensor's body-frame angles to earth-frame angles and position estimate
 //  raw sensor angles stored in _angle_to_target (might be in earth frame, or maybe body frame)
 //  earth-frame angles stored in _ef_angle_to_target
 //  position estimate is stored in _target_pos
 void AC_PrecLand::calc_angles_and_pos(float alt_above_terrain_cm)
 {
     // exit immediately if not enabled
     if (_backend == NULL) {
-        _have_estimate = false;
         return;
     }
 
     // get angles to target from backend
     if (!_backend->get_angle_to_target(_angle_to_target.x, _angle_to_target.y)) {
-        _have_estimate = false;
         return;
     }
 
-    float x_rad;
-    float y_rad;
-
-    if(_backend->get_frame_of_reference() == MAV_FRAME_LOCAL_NED){
```

Comment:
I guess this removes support for the camera to be mounted on a gimbal?  I haven't heard of anyone actually mounting the camera on a gimbal but I just want to be clear that we're removing this functionality.